lib/ios/Ldk.swift

```diff
@@ -937,6 +937,25 @@ class Ldk: NSObject {
             return resolve(Data(res.getValue()?.getA() ?? []).hexEncodedString())
         }
         
+        if let error = res.getError() {
+            switch error.getValueType() {
+            case .APIMisuseError:
+                return handleReject(reject, .start_create_channel_fail, nil, error.getValueAsApiMisuseError()?.getErr())
+            case .ChannelUnavailable:
+                return handleReject(reject, .start_create_channel_fail, nil, error.getValueAsChannelUnavailable()?.getErr())
+            case .FeeRateTooHigh:
+                return handleReject(reject, .start_create_channel_fail, nil, error.getValueAsFeeRateTooHigh()?.getErr())
+            case .InvalidRoute:
+                return handleReject(reject, .start_create_channel_fail, nil, error.getValueAsInvalidRoute()?.getErr())
+            case .IncompatibleShutdownScript:
+                return handleReject(reject, .start_create_channel_fail, nil, "IncompatibleShutdownScript")
+            case .MonitorUpdateInProgress:
+                return handleReject(reject, .start_create_channel_fail, nil, "MonitorUpdateInProgress")
+            @unknown default:
+                handleReject(reject, .start_create_channel_fail, "Unhandled error")
+            }
+        }
+        
         handleReject(reject, .start_create_channel_fail)
```

Comment:
Didn't check if this is 💯 valid Swift, but to make it resemble the Kotlin suggestion I made earlier, this is what the AI generated:
```swift
let failure: Error? = {
    guard let error = res.getError(), 
          case let .err(apiError) = error.getValueType() else { return nil }

    switch apiError {
    case .APIMisuseError(let e): return Error(e.getErr())
    case .ChannelUnavailable(let e): return Error(e.getErr())
    case .FeeRateTooHigh(let e): return Error(e.getErr())
    case .InvalidRoute(let e): return Error(e.getErr())
    case .IncompatibleShutdownScript: return Error("IncompatibleShutdownScript")
    case .MonitorUpdateInProgress: return Error("MonitorUpdateInProgress")
    @unknown default: return nil 
    }
}()

return handleReject(reject, .start_create_channel_fail, nil, failure)
```

Reply:
Not valid, that's not how the LDK bindings return errors